cpp/.clang-tidy

```diff
@@ -39,8 +39,8 @@ Checks:
        -clang-analyzer-optin.core.EnumCastOutOfRange,
        -clang-analyzer-optin.cplusplus.UninitializedObject'
 
-WarningsAsErrors: '*'
-HeaderFilterRegex: '.*cudf/cpp/(src|include|tests).*'
+WarningsAsErrors: ''
+HeaderFilterRegex: '.*cudf/cpp/(src|include).*'
```

Comment:
This change prevents linting of any headers in the tests directory.

Reply:
Would it be clearer to use the `ExcludeHeaderFilterRegex` to *exclude* the `tests` directory?

Reply:
No because there are many more top-level directories in the cpp dir, and because the way that the filtering works the exclusion is layered on top of the inclusion so we're better off using the include to capture the directories that we want and then using the exclude to do more fine-grained filtering.